File: src/TwoSum.java
import java.util.HashMap;
import java.util.Map;

public class TwoSum {

        public int[] twoSum(int[] nums, int target) {

            Map<Integer, Integer> twoSum = new HashMap<>();
            for(int i=0;i<nums.length;i++){
                int temp = target-nums[i];
                if(twoSum.containsKey(temp)){
                    return new int[]{twoSum.get(temp), i};
                }
                twoSum.put(nums[i],i);
            }
            return new int[]{};
        }
}

File: src/two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        sum_map = {}
        for i,val in enumerate(nums):
            if (target-val) in sum_map:
                return [sum_map[target-val],i]
            else:
                sum_map[val] = i